src/java.base/share/classes/jdk/internal/classfile/impl/Util.java

```diff
@@ -249,16 +253,11 @@ static void writeList(BufWriterImpl buf, List<Writable> list) {
     }
 
     public static int slotSize(ClassDesc desc) {
-        return switch (desc.descriptorString().charAt(0)) {
-            case 'V' -> 0;
-            case 'D','J' -> 2;
-            default -> 1;
-        };
+        return isDoubleSlot(desc) ? 2 : 1;
```

Comment:
No, I meant for this to be implemented in a new method, `paramSlotSize`, then call that instead of `slotSize` in the places where that's safe.